pkg/sqlite/load.go

```diff
@@ -806,7 +806,11 @@ func (s *SQLLoader) updatePackageChannels(tx *sql.Tx, manifest registry.PackageM
 		// because this is caught by primary key of operatorbundle table
 
 		channelEntryCSV, err := s.getCSV(tx, c.CurrentCSVName)
-
```

Comment:
`updatePackageChannels()` should only be called in the `opm registry add` flow. I'm curious as to why this function is being called in the `initializer` flow in this instance - if you look at `AddPackageChannels` it only calls this function when adding a packageName to the database fails. Could you investigate the root of that failure?

Reply:
@gallettilance looks like when if there's a problem with unmarshaling the CSV, the CSV is not added to the [`operatorBundle` table](https://github.com/operator-framework/operator-registry/blob/master/pkg/sqlite/migrations/000_init.go#L18) and the [`channel` table](https://github.com/operator-framework/operator-registry/blob/master/pkg/sqlite/migrations/000_init.go#L28), i.e it is skipped. Therefore when we try to add a new [package](https://github.com/operator-framework/operator-registry/blob/master/pkg/sqlite/migrations/000_init.go#L23) which references the `channel` table, the Foreign Key constraint fails, as a result adding a package fails, and therefore, it tries to update the channel.